source/usage-examples/insertOne.txt

```diff
@@ -0,0 +1,91 @@
+.. _csharp-insert-one:
+
+=================
+Insert a Document
+=================
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+You can insert a single document into a ``MongoCollection`` by using the
+`InsertOne() <{+api-root+}/M_MongoDB_Driver_IMongoCollection_1_InsertOne.htm>`__
+method.
+
+Examples
+--------
+
+These examples use the following ``Restaurant`` class as a model:
+
+.. literalinclude:: ../includes/code-examples/FindOne.cs
+   :start-after: start-model
+   :end-before: end-model
+   :language: csharp
+   :copyable:
+   :dedent:
+
+Insert a Document
+~~~~~~~~~~~~~~~~~
+
+.. tabs::
+
+   .. tab:: Asynchronous
+      :tabid: insert-one-async
+      
+      The following example asynchronously inserts a document into 
+      the ``restaurants`` collection.
+
+      .. literalinclude:: ../includes/code-examples/InsertOneAsync.cs
+         :start-after: start-insert-one-async
+         :end-before: end-insert-one-async
+         :language:  csharp
+         :copyable:
+         :dedent:
+
+   .. tab:: Synchronous
+      :tabid: insert-one-sync
+
+      The following example asynchronously inserts a document into the ``restaurants``
+      collection.
+
+      .. literalinclude:: ../includes/code-examples/InsertOne.cs
+         :start-after: start-insert-one
+         :end-before: end-insert-one
+         :language:  csharp
+         :copyable:
+         :dedent:
+
+Expected Result
+~~~~~~~~~~~~~~~
+
+After running the :ref:`full example <insert-runnable-example>`, the ``InsertOne()``
+method inserts the document, and the :ref:`Find() <csharp-find-one>` method returns
+the newly inserted document. The output is as follows:
+
+.. code-block:: none
+
+   Inserting a document...
+   Document Inserted: { "_id" : ObjectId("631fa292c44ae48b23d88eba"), "name" : "Mongo's Pizza", "restaurant_id" : "12345", "cuisine" : "Pizza", "address" : { "_t" : "MongoDB.Bson.BsonDocument, MongoDB.Bson", "_v" : { "street" : "Pizza St", "zipcode" : "10003" } }, "borough" : "Manhatten", "grades" : [{ "_t" : "MongoDB.Bson.BsonDocument, MongoDB.Bson", "_v" : { } }] }
```

Comment:
issue: The ObjectId is pretty much guaranteed to be different; consider truncating (...).